src/components/SingleInputField.tsx

```diff
@@ -0,0 +1,45 @@
+import { FormControl, FormHelperText, FormLabel, Input } from '@chakra-ui/react';
+import { FormikProps } from 'formik';
+
+export const SingleInputField = <T,>({
+  accessor,
+  formik,
+  type,
+  min,
+  max,
+  step,
+  text,
+  valueAsString,
+}: {
+  accessor: keyof T & string;
+  formik: FormikProps<T>;
+  type: string;
+  text: string;
+  max: number | string;
+  min: number | string;
+  step: number | string;
+  valueAsString: boolean;
+}) => {
+  const checkTypeOfValue = valueAsString ? 'string' : 3;
```

Comment:
to sprawia że value potem jest typu 'string' albo 3. Nie do końca rozumiem tę linię

Reply:
dokladnie, a pozniej jest to tak:

       value={formik.values[accessor] as typeof checkTypeOfValue}
       
       nie wiedzialem jak to lepiej rozpisac

Reply:
skąd tam się to 3 bierze bo jak już to chyba string lub number powinno być?

Reply:
3 bierze sie z czapy, po prostu musze wstawic liczbe aby wstawic typeof liczby i uzyskac number